### PR TITLE
Update webhook dedicated page

### PR DESCRIPTION
**Pull request by arun-singh**
Updated webhooks to be more like https://argyle.com/docs/api-guide/webhooks.

Some things we can't add yet like wildcards for event subscriptions or the static IPs to whitelist. 

Didn't add the code snippet for listening and verifying.

### CODE REVIEW

File: api-reference/webhooks.mdx [hunk 1]
```diff
@@ -39,15 +39,37 @@ curl --request POST \
 - **secret**: (Optional) A secret for secure webhook verification.
 
 ### Verifying Webhooks
-Use the provided `secret` to validate incoming webhook signatures, ensuring data integrity and authenticity.
+If the optional `secret` was provided in the webhook subscription, it can be used to validate incoming webhook signatures, ensuring data integrity and authenticity.
```
Comment (luchesar) [inline api-reference/webhooks.mdx:42]: Is it worth mentioning similar higlighted fact as:
image

File: api-reference/webhooks.mdx [hunk 2]
```diff
@@ -39,15 +39,37 @@ curl --request POST \
...
+              "timestamp": "2024-04-04T12:00:00.00Z"
+            }
+```
+
+The signature to check against would be:
+
+```
+"X-Teal-Signature" : "a30540779107a19069257432b775b74b16b32214616638fae2e6027a41a3f2dfb08f44daf3862c335d08fb83501fc769f73d49a1cb137f96f31c6a7db412c197"
+```
+
```
Comment (luchesar) [inline api-reference/webhooks.mdx:63]: Is it worth adding the signature verification code example:
```js
const http = require('http');
const crypto = require('crypto');

const serverPort = 80
const requestListener = function (req, res) {
    const client_secret = 'my little secret';
    var requestBody = '';
    req.on('readable', () => {
        var read = req.read()
        if(read != null) {
        requestBody += read;
        }
    });
    req.on('end', () => {
        // Make sure to use the raw requestBody when validating the signature.
        signature = crypto.createHmac('sha512',client_secret).update(requestBody).digest('hex');
        console.log('Signature:' + signature);
        res.writeHead(200);
        res.end(signature);
    })
}
const server = http.createServer(requestListener);
server.listen(serverPort);
```

**Review by luchesar — CHANGES_REQUESTED** [review]

File: api-reference/webhooks.mdx (hunk 1 above)
Comment (arun-singh) [reply api-reference/webhooks.mdx:42]: We have 

`Webhooks are system-wide, not limited to specific users or accounts, enhancing your application's responsiveness.` 

above but have changed it to:

`Webhooks are system-wide meaning subscriptions are not limited to specific users or accounts. For example, if any user updates their payroll with the `user-payroll-submitted` subscription enabled, 
a webbook will be delivered with the respective user information enclosed.`

File: api-reference/webhooks.mdx (hunk 2 above)
Comment (arun-singh) [reply api-reference/webhooks.mdx:63]: THink its slightly overkill for now

**Review by luchesar — APPROVED** [review]
LGTM

**Pull request merged**